extra_data/keydata.py

```diff
@@ -169,6 +169,42 @@ def data_counts(self, labelled=True):
             idxs = np.argsort(train_ids)
             return counts[idxs]
 
+    def single_value(self, rtol=1e-5, atol=0.0, reduce_by='median'):
+        """Retrieve a single reduced value if within tolerances.
+
+        The relative and absolute tolerances *rtol* and *atol* work the
+        same way as in ``numpy.allclose``. The default relative tolerance
+        is 1e-5 with no absolute tolerance. The data for this key is compared
+        against a reduced value obtained by the method described in *reduce_by*.
+
+        This may be a callable taking the key data, the string value of a
+        global symbol in the numpy packge such as 'median' or 'first' to use
+        the first value encountered. By default, 'median' is used.
+
+        If within tolerances, the reduced value is returned.
+        """
+
+        data = self.ndarray()
```

Comment:
Do we want to check for a length 0 array and raise our own error? At present, you'd get `nan` with a RuntimeWarning for median/mean, and an IndexError for 'first'.

Reply:
Oh good catch, that is definitely not consistent. I could see two options:
* Raise an error
* Return `None`

I gravitate towards the first, but then again regular read methods also don't throw an error an `None` could not be returned in any other case. It boils down to EAFP vs LBYL.

Reply:
Hmmm, any preference on the exception type? A simple one might be `RuntimeError`, but that's maybe a bit too generic. Functions like `np.max()` throw a `ValueError` when called on an empty array. I don't see any fitting type in `extra_data.exceptions`. Maybe a `NoDataError` to be precise?

Reply:
Python built in `min()` and `max()` also throw ValueError. The [`statistics` module](https://docs.python.org/3/library/statistics.html) defines its own StatisticsError, inheriting from ValueError. `functools.reduce` is something of an outlier, throwing TypeError.

So I'd either use ValueError or subclass it as NoDataError. I don't have a particular preference between those options.